lib/src/components/Tab/Tab.tsx

```diff
@@ -0,0 +1,86 @@
+import { assignInlineVars } from "@vanilla-extract/dynamic";
+import { useEffect, useState } from "react";
+import { Badge } from "../Badge";
+import {
+  badgeStyles,
+  contentStyles,
+  iconStyles,
+  tabHeaderContainerStyles,
+  tabRecipe,
+  tabVars,
+} from "./tab.css";
+import { TabsObjectProps, TabProps } from "./tab.types";
+
+export const Tab = ({
+  defaultValue,
+  value,
+  color = "#2F80ED",
+  background = "#d7e9ff",
+  onChange = () => {},
+  grow = false,
+  height = "40px",
+  tabs,
+  style,
+  className,
+  children,
+  ...nativeProps
+}: TabProps) => {
+  const [selectedTab, setSelectedTab] = useState<TabsObjectProps>(
+    {} as TabsObjectProps
+  );
+
+  useEffect(() => {
+    const comparator = value || defaultValue;
+    const tab = tabs.find((arr) => arr.value === comparator);
+    tab && setSelectedTab(tab);
+  }, [defaultValue, tabs, value]);
+
+  return (
+    <div>
+      <div
+        style={assignInlineVars({
+          [tabVars.height]: height,
+        })}
+        className={`${tabHeaderContainerStyles}`}
+      >
+        {tabs.map((arr, ind) => {
```

Comment:
same as above `arr` is generic name. in the future the purpose will be difficult to understand as complexity of code increases